woocommerce/woocommerce
# Accessibility enhancement for the whole shop accounts section

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

If you the above steps in `trunk` the voice ready will not read out the errors.



### Changes proposed in this Pull Request:




This PR solves the Accessibility issues for the whole shop accounts section and closes several issues reported by the Author.

- As part of the fix I wrote a script that focus on the message if a notice is present for both error and failures after the full page reload.
- Since this only affects the accounts section for now, so I am loading the script only on that route.

Closes #43635 , #43659 , #43636 , https://github.com/woocommerce/woocommerce/issues/46559



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use the Safari browser for this testing if you are on a Mac or Chrome for Windows.
2. Visit `accounts` section of the shop.
3. Make sure you are logged out.
4. Enable Voice reader (On Mac, press `CMD + F5`). There must be other keys for other OS.
5. Now try to log in without entering the username or password.
6. The reader should read out the error message completely.
7. Next -> The voice reader should read out the error or success message in the following scenarios. 
8. Registering a new account without an email ID or the wrong email ID.
9. Not submitting the email or username on the forgot password page should read out the error.
10. Changing account details after login in the account section, try submitting the form without any data, voice reader should read out multiple errors.
11. Changing the account details with correct data should read out the success message.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
